src/mbgl/renderer/renderer_impl.cpp

```diff
@@ -140,6 +141,13 @@ void Renderer::Impl::render(const RenderTree& renderTree) {
 
     parameters.depthRangeSize = 1 - (layerRenderItems.size() + 2) * parameters.numSublayers * parameters.depthEpsilon;
 
+    // Drawables
+    {
+        for (const auto &drawable : layerDrawables) {
```

Comment:
And here iterate on renderTree.getDrawables() directly to avoid a copy of the render tree's vector

Reply:
Some projects avoid that due to the possibility of the caller keeping the reference too long or otherwise abusing it, but if you're okay with it returning a const ref is my preference as well.

Reply:
At least in this case, the lifetime of the render tree is coupled directly with the renderer, so temporary references here are safe. Most important, making copies on the hot-path should only be done in cases where concurrency is required and atomics or locks are a worse option compared to the copy.